ikasaneip/topology/src/main/java/org/ikasan/topology/metadata/JsonFlowMetaDataProvider.java

```diff
@@ -0,0 +1,223 @@
+package org.ikasan.topology.metadata;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import org.ikasan.spec.component.endpoint.Broker;
+import org.ikasan.spec.component.endpoint.Consumer;
+import org.ikasan.spec.component.endpoint.Producer;
+import org.ikasan.spec.component.filter.Filter;
+import org.ikasan.spec.component.routing.MultiRecipientRouter;
+import org.ikasan.spec.component.routing.SingleRecipientRouter;
+import org.ikasan.spec.component.splitting.Splitter;
+import org.ikasan.spec.component.transformation.Converter;
+import org.ikasan.spec.component.transformation.Translator;
+import org.ikasan.spec.configuration.ConfiguredResource;
+import org.ikasan.spec.flow.Flow;
+import org.ikasan.spec.flow.FlowConfiguration;
+import org.ikasan.spec.flow.FlowElement;
+import org.ikasan.spec.metadata.FlowElementMetaData;
+import org.ikasan.spec.metadata.FlowMetaData;
+import org.ikasan.spec.metadata.FlowMetaDataProvider;
+import org.ikasan.spec.metadata.Transition;
+import org.ikasan.topology.metadata.model.FlowElementMetaDataImpl;
+import org.ikasan.topology.metadata.model.FlowMetaDataImpl;
+import org.ikasan.topology.metadata.model.TransitionImpl;
+
+import java.util.Map;
+
+/**
+ * Implementation of FlowMetaDataProvider that provides a JSON representation of a flow.
+ *
+ * @author Ikasan Development Team
+ */
+public class JsonFlowMetaDataProvider implements FlowMetaDataProvider<String>
+{
+
+    @Override
+    public String describeFlow(Flow flow)
+    {
+        String result;
+
+        try
+        {
+            FlowConfiguration configuration = flow.getFlowConfiguration();
+            flow.getName();
+
+            FlowMetaDataImpl flowMetaData = new FlowMetaDataImpl();
+            flowMetaData.setName(flow.getName());
+            flowMetaData.setConsumer(describeFlowElement(flowMetaData, configuration.getConsumerFlowElement()));
+
+            if(flow instanceof ConfiguredResource)
+            {
+                flowMetaData.setConfigurationId(((ConfiguredResource) flow).getConfiguredResourceId());
+            }
+
+            ObjectMapper mapper = new ObjectMapper();
+
+            result = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(flowMetaData);
+        }
+        catch (Exception e)
+        {
+            throw new RuntimeException("Exception has occurred creating flow meta data json!", e);
+        }
+
+        return result;
+    }
+
+    @Override
+    public FlowMetaData deserialiseFlow(String metaData)
+    {
+        FlowMetaDataImpl result;
+
+        try
+        {
+            ObjectMapper mapper = new ObjectMapper();
```

Comment:
the mapper should be initialised in constructor once. there is no need for creating new mapper every time.